tests/migration_test.go

```diff
@@ -1766,11 +1766,22 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 						Expect(err).ToNot(HaveOccurred(), "Should successfully get new VMI")
 						vmiPod := tests.GetRunningPodByVirtualMachineInstance(newvmi, tests.NamespaceTestDefault)
 
-						if newvmi.Status.Interfaces[0].IP == vmiPod.Status.PodIP {
-							return nil
+						if newvmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP {
+							return fmt.Errorf("IP field doesnt match PodIP")
 						}
 
-						return fmt.Errorf("interface not updated with latest IP")
+						if len(newvmi.Status.Interfaces[0].IPs) != len(vmiPod.Status.PodIPs) ||
+							len(newvmi.Status.Interfaces[0].IPs) == 0 {
+							return fmt.Errorf("Status.Interfaces len is zero or not equal PodIPs len")
```

Comment:
Isn't assuring `len != 0` ? (you've checked previously that `newvmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP` )

I mean that - afaiu - if you have `PodIP` it means you'll have **at least** one IP in `pod.Status.PodIPs`.

Reply:
Let's use individual checks with specific errors instead of `||` them.

Reply:
> Isn't assuring `len != 0` ? (you've checked previously that `newvmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP` )
> 
> I mean that - afaiu - if you have `PodIP` it means you'll have **at least** one IP in `pod.Status.PodIPs`.

what if there arent ips, would PodIP exists or not ?
(kubevirt doesnt support it but atm at least, what about the future?)
I think that even if its redundant, its a simple sanity that i would suggest to keep

Reply:
> > Isn't assuring `len != 0` ? (you've checked previously that `newvmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP` )
> > I mean that - afaiu - if you have `PodIP` it means you'll have **at least** one IP in `pod.Status.PodIPs`.
> 
> what if there arent ips, would PodIP as a field exists or not (i guess you right and it wont be just empty existing field) ? - in this case len can be zero.

That is against [k8s networking paradigm](https://kubernetes.io/docs/concepts/cluster-administration/networking/#the-kubernetes-network-model); a pod has its own IP address. 

And we control the implementation on kubevirt.

> (kubevirt doesnt support it but atm at least, what about the future?)
> I think that even if its redundant, its a simple sanity that i would suggest to keep

The issue comes when all those 'simple' sanities piled up distract the reader.

Reply:
exactly now on other PR this same "check IPs" is not empty caught a bug (more details to come), i really think we shouldnt omit it please.

Reply:
Please elaborate; what bug did this catch ?

Reply:
As i wrote "more details to come" (when i find it self contained to summarize please).

But meanwhile, one possible reason for example:
since the loop is range vmiPod.Status.PodIPs, what if the VM is still updating and the `newvmi.Status.Interfaces[0].IPs` isnt full yet, there will be a panic index out of range. 

When you run a test, and get vmi -o yaml in a loop, you will see it takes time for all the fields to get updated,
and here its a migration test that might have it as well, because those update comes in events in the core logic.

Reply:
moved to function and did what Quique asked (split)